kratos/tests/cpp_tests/containers/test_pointer_vector_set.cpp

```diff
@@ -443,6 +537,51 @@ KRATOS_TEST_CASE_IN_SUITE(TestPointerVectorSetInsert9, KratosCoreFastSuite)
     KRATOS_EXPECT_EQ(&*(itr++), &*elements[28]);
 }
 
+KRATOS_TEST_CASE_IN_SUITE(TestPointerVectorSetInsert10, KratosCoreFastSuite)
+{
+    PointerVectorSet<Element, IndexedObject, std::less<IndexType>, std::equal_to<IndexType>, std::shared_ptr<Element>> test_container;
+
+    std::vector<std::shared_ptr<Element>> elements;
+    for (IndexType i = 0; i < 50; ++i) {
+        elements.push_back(Kratos::make_shared<Element>(i + 1));
+    }
+
+    std::vector<std::shared_ptr<Element>> tmp;
+    tmp.push_back(elements[5]);
+    tmp.push_back(elements[28]);
+    tmp.push_back(elements[2]);
+    tmp.push_back(elements[5]);
+    tmp.push_back(elements[28]);
+    tmp.push_back(elements[1]);
+    tmp.push_back(elements[25]);
+
+    // following will change the original tmp
+    test_container.insert(std::move(tmp));
+
+    // now we check for the order
+    auto itr_temp = tmp.begin();
+    KRATOS_EXPECT_EQ(tmp.size(), 7);
+    KRATOS_EXPECT_EQ((**(itr_temp++)).Id(), 2);
+    KRATOS_EXPECT_EQ((**(itr_temp++)).Id(), 3);
+    KRATOS_EXPECT_EQ((**(itr_temp++)).Id(), 6);
+    KRATOS_EXPECT_EQ((**(itr_temp++)).Id(), 26);
+    KRATOS_EXPECT_EQ((**(itr_temp++)).Id(), 29);
+
+    // we cannot check for the following because it will always pointing to the nullptr
+    // KRATOS_EXPECT_EQ((**(itr_temp++)).Id(), 6);
+    // but this we can check for the following
+    KRATOS_EXPECT_EQ((*(itr_temp++)), nullptr);
+    KRATOS_EXPECT_EQ((**(itr_temp++)).Id(), 29);
```

Comment:
```suggestion
```

this is undefined behavior. You mustn't rely on the state of `tmp` after moving it into `test_container`.

Besides, the fact that the item following unique'd range's end is `nullptr` is a result of the standard library's implementation of `std::unique`, which might change in the future.

Reply:
yes, tmp simply "doesn't exist" once you moved